### PR TITLE
Add new docs for the event manager project.

### LINKED ISSUE #4: Add documentation

### PR DESCRIPTION
**Pull request by jwage**
Fixes #4

### CODE REVIEW

File: docs/en/index.rst [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
-   :depth: 2
-   :glob:
+Getting Help
+------------
 
-   *
+If this documentation is not helping to answer questions you have about the
+Doctrine DBAL, don't panic. You can get help from different sources:
+
+-  Gitter chat room `#doctrine/event-manager <https://gitter.im/doctrine/event-manager>`_
```
Comment (Majkl578) [inline docs/en/index.rst]: Do we actually have gitter rooms per repo?

File: docs/en/index.rst [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+-  On `Stack Overflow <http://stackoverflow.com/questions/tagged/doctrine-event-manager>`_
+-  The `Doctrine Mailing List <http://groups.google.com/group/doctrine-user>`_
+-  Report a bug on `GitHub <https://github.com/doctrine/event-manager/issues>`_.
+
+Getting Started
+---------------
+
+The best way to get started is with the :doc:`Introduction <reference/index#introduction>` section
+in the documentation. Use the sidebar to browse other documentation for the Doctrine PHP DBAL.
```
Comment (Majkl578) [inline docs/en/index.rst]: typo: DBAL

File: docs/en/reference/index.rst [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+        /** @var bool */
+        public $preFooInvoked = false;
+
+        /** @var bool */
+        public $postFooInvoked = false;
+
+        public function __construct(EventManager $eventManager)
+        {
+            $eventManager->addEventListener(array(self::preFoo, self::postFoo), $this);
```
Comment (Majkl578) [inline docs/en/reference/index.rst]: `[]` instead of `array()`

File: docs/en/reference/index.rst [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    $eventManager->dispatchEvent(TestEvent::postFoo);
+
+Removing Event Listeners
+========================
+
+You can easily remove a listener with the ``removeEventListener()`` method.
+
+.. code-block:: php
+
+    $eventManager->removeEventListener(array(TestEvent::preFoo, TestEvent::postFoo), $testEvent);
```
Comment (Majkl578) [inline docs/en/reference/index.rst]: `[]` instead of `array()`

File: docs/en/reference/index.rst [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        public $preFooInvoked = false;
+
+        public function preFoo() : void
+        {
+            $this->preFooInvoked = true;
+        }
+
+        public function getSubscribedEvents() : array
+        {
+            return array(TestEvent::preFoo);
```
Comment (Majkl578) [inline docs/en/reference/index.rst]: `[]` instead of `array()`

File: docs/en/index.rst (hunk 1 above)
Comment (jwage) [reply docs/en/index.rst]: Yes, I have been creating them. I will create it for this project.

File: docs/en/index.rst (hunk 2 above)
Comment (jwage) [reply docs/en/index.rst]: I don't see the typo?

Comment (Majkl578) [reply docs/en/index.rst]: Sorry, it's not DBAL, Its event manager.

**Pull request merged**